From fe6db4588732d915d85f0ef4d0c1b49d4feaa9ed Mon Sep 17 00:00:00 2001
From: Dawei Chen <dchen@silexamerica.com>
Date: Wed, 2 Aug 2017 09:52:00 -0700
Subject: [PATCH] Fix GATT Glib-WARNING Fix Glib-WARNING**:. Invalid file
 descriptor.

---
 attrib/interactive.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/attrib/interactive.c b/attrib/interactive.c
index 171b957..2bf854e 100644
--- a/attrib/interactive.c
+++ b/attrib/interactive.c
@@ -180,10 +180,11 @@ static void disconnect_io()
 	attrib = NULL;
 	opt_mtu = 0;
 
+#if 0 /* silex change */
 	g_io_channel_shutdown(iochannel, FALSE, NULL);
 	g_io_channel_unref(iochannel);
 	iochannel = NULL;
-
+#endif
 	set_state(STATE_DISCONNECTED);
 }
 
