From f887913ea24e7f75239cfe8e808323c8691c4550 Mon Sep 17 00:00:00 2001
From: Dawei Chen <dchen@silexamerica.com>
Date: Wed, 2 Aug 2017 09:52:00 -0700
Subject: [PATCH] Fix GATT Glib-WARNING Fix Glib-WARNING**:. Invalid file
 descriptor.

---
 attrib/interactive.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/attrib/interactive.c b/attrib/interactive.c
index 7d4786a..ca585fe 100644
--- a/attrib/interactive.c
+++ b/attrib/interactive.c
@@ -192,10 +192,11 @@ static void disconnect_io()
 	attrib = NULL;
 	opt_mtu = 0;
 
+#if 0 /* silex change */
 	g_io_channel_shutdown(iochannel, FALSE, NULL);
 	g_io_channel_unref(iochannel);
 	iochannel = NULL;
-
+#endif
 	set_state(STATE_DISCONNECTED);
 }
 
-- 
2.7.4

